### ISSUE TITLE
Decompiler produces an (illegal) switch expression where it doesn't makes sense

### ISSUE THREAD

**Issue by Kroppeb** [QuiltMC/quiltflower#161]
Source
```java
  public void test(Object a) {
    switch (10000) {
      default: {
        Object vvv26 = null;
        while (vvv26 == null) {
        }
      }
      case -79:
        long vvv36 = -5851592879203823622L;
      case 320:
        if (a != null) {
          return;
        }
    }
    System.out.println("Hi");
  }
 ```
 Result:
 ```java
    public void test(Object a) {
      long var4 = switch(10000) {
         case 320 -> {
         }
         default -> {
            Object vvv26 = null;

            while(vvv26 == null) {
            }
         }
         case -79 -> -5851592879203823622L;
      };
      if (a == null) {
         System.out.println("Hi");
      }
   }
```

**Comment by SuperCoder7979** [comment]
Larger case: https://gist.github.com/SuperCoder7979/9d7d9ad910edcdb03e87c878376515be

**Comment by Kroppeb** [comment]
Even larger case: [https://gist.github.com/Kroppeb/9bebf23637107b0bbf210782cd31155f](https://gist.github.com/Kroppeb/9bebf23637107b0bbf210782cd31155f)

**Comment by SuperCoder7979** [comment]
Should be fixed by e37b6c7e
